Is cross compiling CoreCLR for apple silicon possible on a x64 MacOS machine?

More of an inquiry than anything. I attempted: `./build.sh -a arm64` on my macbook

I saw the following error in the build log:
```
runtime/src/coreclr/pal/src/synchmgr/synchmanager.cpp:4629:6: error: "Don't know how to get hi-res current time on this platform"
      #error "Don't know how to get hi-res current time on this platform"
```

Seems like neither `HAVE_WORKING_CLOCK_GETTIME` or `HAVE_WORKING_GETTIMEOFDAY` are being defined when building this way?

Potentially relevant system info:
Xcode version: 13.1
OS: Monterey 12.0.1
I've ran `brew bundle --no-lock --file eng/Brewfile` already

--- a/docs/workflow/building/coreclr/cross-building.md
+++ b/docs/workflow/building/coreclr/cross-building.md
@@ -19,6 +19,12 @@ Build using "arm" as the architecture. For example:
 
     C:\runtime> src\coreclr\build-runtime.cmd -arm -debug
 
+Cross Compilation for ARM64 on macOS
+================================================
+
+The toolset required to do native compilation described in [macOS requirements](https://github.com/dotnet/runtime/blob/main/docs/workflow/requirements/macos-requirements.md) has cross compilation capabilities with no additional installs required. The only needed change is to use the `-cross` flag in the build scripts. For example, a release CoreCLR runtime that targets ARM64 can be built using the following commandline from the root of the repo:
+
+    ./build.sh -subset clr.runtime -c release -cross -arch arm64 /p:CrossBuild=true
 
 Cross Compilation for ARM, ARM64 or x86 on Linux
 ================================================

